### DISCUSSION TITLE
BUG: Grapesjs bootstrap integration

### DISCUSSION THREAD

**Discussion by srivatsan2607** [GrapesJS/grapesjs#4670] [Q&A]
With the latest version of grapesjs v0.20.1, trying to import GrapesJS Bootstrap v4 Blocks Plugin produces the following error which results in not rendering of bootstrap components.

TypeError: Cannot set property defaults of #<n> which has only a getter
    at grapes.min.js:2:1
    at Module.It (grapes.min.js:2:1)
    at Function.$ [as extend] (grapes.min.js:2:1)
    at tt (grapesjs-blocks-bootstrap4.min.js:2:1)
    at nn (grapesjs-blocks-bootstrap4.min.js:2:1)
    atgrapesjs-blocks-bootstrap4.min.js:2:1
    at grapes.min.js:2:1
    at Array.forEach (<anonymous>)
    at Object.init (grapes.min.js:2:1)
    at Editor.js:45:1

**Comment by hyuuks** [comment]
im sure the error are not from grapesjs, read the docs, change on version changed on codes, and you should include your codes in your question

**Comment by artf** [comment]
Probably the plugin uses the old version of API to register components, the breaking change was posted on release notes: https://github.com/artf/grapesjs/releases/tag/v0.20.1

**Comment by jcenlo** [comment]
@hyuuks  Like the notes say if you want to use in the new version of grapesjs , my solution is edit the plugin 

Remove this code
<img width="681" alt="Screenshot 2022-10-22 at 01 41 23" src="https://user-images.githubusercontent.com/1485745/197305677-c9265210-c4ab-47f2-90c4-1a3d1cf696c1.png">

and you should have something like 
<img width="496" alt="Screenshot 2022-10-22 at 01 45 12" src="https://user-images.githubusercontent.com/1485745/197305734-213ae3f1-25de-43e1-8075-4e974c3cf9f8.png">

I am not full test this but at least the error dissapear

**Reply by jcenlo** [reply]
@srivatsan2607 ,i just improve the result, this is the example of the image component,

i feel a lot of people here is working in a bootstrap 5 grapesjs plugin, ¿Should we do a community plugin?, is a pain in the ass do it alone
```JavaScript
import imageIcon from "raw-loader!../../icons/image-solid.svg";

export const ImageBlock = (bm, label) => {
    bm.add("bs-image", {
        label: `
            ${imageIcon}
            <div>${label}</div>
        `,
        category: "Media",
        content: {
            type: "image",
        },
    });
};

export default (domComponent) => {
    const img_src_default = "https://dummyimage.com/200x200/999/222";
    const imageType = domComponent.getType("image");

    const type = "image";

    domComponent.addType(type, {
        extend: "image",

        isComponent: function (el) {
            if (el && el.tagName === "IMG") {
                return { type: type };
            }
        },

        model: {
            defaults: {
                ...imageType.model.prototype.defaults,
                tagName: "img",
                resizable: 1,
                attributes: {
                    src: img_src_default,
                },
                classes: ["img-fluid"],
                traits: [
                    {
                        type: "text",
                        label: "Source (URL)",
                        name: "src",
                    },

                    ...imageType.model.prototype.defaults.traits,
                ],
            },
        },
    });
};

```
